### DISCUSSION TITLE
Create a dependency using an async function of another dependency

### DISCUSSION THREAD

**Discussion by phi161** [pointfreeco/swift-dependencies#128] [Q&A]
I have dependency `DependencyA` which needs a `Driver` to perform its work.
The problem is that to create a driver, I need the result of some other dependency (let's say `Settings`), which is an async call:

```
extension DependencyA: DependencyKey {
    public static var liveValue: DependencyA = {
        @Dependency(\.settings) var settings
        let isOn = try await settings.isOn() // error, this is an async function!
        let driver = Driver(isOn: isOn)
        return make(driver: driver)
    }()

    static func make(driver: Driver) -> Self {
        DependencyA(
            endpoint1: {
                driver.doSomething()
            },
        // ... other endpoints
```

Is it possible to have this sort of setup?

**Comment by ammerzon** [comment]
I experience the same problem right now where I need to sync the credentials of a server in dependency A, which is later used in dependency B.

**Comment by natemann** [comment]
Do you actually need the the option to override your settings dependency in your live value of DependencyA? Just simple use your live value of settings directly in DependencyA.

I have to call UserDefaults within 2-3 of my Dependencies. I simply just call UserDefaults directly instead of using the dependency value of UserDefaults.
